File: src/main/java/org/flossware/jnexus/RepoRecord.java
package org.flossware.jnexus;

/**
 * Represents a repository component record from Nexus.
 * <p>
 * This record encapsulates the essential information about a component stored
 * in a Nexus repository, including its unique identifier, file size, and path.
 * </p>
 *
 * @param id       the unique identifier of the component in Nexus
 * @param fileSize the size of the component file in bytes (supports files up to ~8 exabytes)
 * @param path     the full path to the component artifact
 * @author sfloess
 * @since 1.0
 */
public record RepoRecord(String id, long fileSize, String path) {

}

File: jnexus-ios/Shared/Models/RepoRecord.swift
//
//  RepoRecord.swift
//  JNexus
//
//  Basic repository component record
//  Mirrors the Java RepoRecord from jnexus-core
//

import Foundation

/// Basic component record with essential fields
struct RepoRecord: Codable, Identifiable, Hashable {
    /// Unique component identifier
    let id: String

    /// File size in bytes
    let fileSize: Int64

    /// Component path in repository
    let path: String

    /// CodingKeys for JSON serialization
    enum CodingKeys: String, CodingKey {
        case id
        case fileSize
        case path
    }
}
